buildchow/templates/defaults.yaml

```diff
@@ -4,35 +4,27 @@
     default-jdk: jdk1.8
     default-maven-version: maven-3.2.5
     default-remote: origin
-    jvm-memory: "-Xms512m -Xmx1024m -XX:PermSize=128m -XX:MaxPermSize=256m"
+    default-releng-publish-zip: "org.jboss.tools.releng:jbosstools-releng-publish:4.3.0.CR1-SNAPSHOT:zip"
+    default-daysToKeep: 10
+    default-numToKeep: 10
+    default-artifactDaysToKeep: 10
+    default-artifactNumToKeep: 10
+    default-nodes: "(RHEL6||RHEL7||beaker||jboss-prod)&&!ia64&&!ppc64&&!rhts"
+    # since we use JDK 8 now, no need for these:  -XX:PermSize=128m -XX:MaxPermSize=256m"
+    jvm-memory: "-Xms512m -Xmx1536m"
     tests-folder: tests # for some it is all-tests
     component-responsible: ""
     scm-schedule: H H/8 * * * # every 8 hours starting a random hour. Removes need for each job to configure it (http://htmlpreview.github.io/?https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.html)
-    download-cache:  "$JENKINS_HOME/static_build_env/jbds/download-cache"
-    maven-settings: "$JENKINS_HOME/../.m2/local-settings.xml" # need more reliable location
+    # these paths only exist on NFS-mounted slaves in the JBoss Jenkins cluster in mw.lab.eng.bos.redhat.com, which have a hudson user on them
+    # some slaves are configured with JENKINS_HOME = /qa/hudson_master/hudson_home/hudson_workspace, which is NOT the path we want here 
+    HUDSON_USER_HOME: "/home/hudson"
+    download-cache:  "{HUDSON_USER_HOME}/static_build_env/jbds/download-cache"
```

Comment:
1. JENKINS_HOME is not a defined variable on mv.lab.eng.bos slaves. Not sure where you found that. 

Recommend using either $HOME (which will be `/qa/services/hudson` or `/home/hudson` depending on the slave -- see references below - at item 4), or perhaps a combination of:

* $HUDSON_STATIC_ENV ($HOME/static_build_env) for the path to `static_build_env/jbds/` and 
* $HUDSON_HOME ($HOME/hudson_workspace) for the path to `.m2/settings.xml`. 

2. I'm not sure we even need `.m2/local-settings.xml`, which is why I commented it out -- you put that in, but I don't see a job config.xml which calls it. Maybe it's a leftover from an earlier time?

3. My point with setting HUDSON_USER_HOME: "/home/hudson" and wanting to externalize that into the server config, is that depending on the slave, you could set a different value. Thus for SOME Jenkins slaves it would be "/home/hudson" and for virt slaves it could be "/qa/services/hudson".

4. If you need examples of the differences in configs, see these:

http://jenkins.mw.lab.eng.bos.redhat.com/hudson/computer/vmg39-rhel6-x86_64/systemInfo
http://jenkins.mw.lab.eng.bos.redhat.com/hudson/computer/dspvmg03-rhel6-x86_64/systemInfo
vs.
http://jenkins.mw.lab.eng.bos.redhat.com/hudson/computer/dev128-virt2/systemInfo
http://jenkins.mw.lab.eng.bos.redhat.com/hudson/computer/dev125-virt2/systemInfo
http://jenkins.mw.lab.eng.bos.redhat.com/hudson/computer/dev127-virt1/systemInfo
http://jenkins.mw.lab.eng.bos.redhat.com/hudson/computer/dev126-virt1/systemInfo

5. The reason I chose HUDSON_USER_HOME instead of just HOME is that you might not be running Jenkins locally as your own user, but rather a different one; thus YOUR $HOME != $HUDSON_USER_HOME